clients/pkg/promtail/targets/kafka/target.go

```diff
@@ -80,13 +89,28 @@ func (t *Target) run() {
 		if len(lbs) > 0 {
 			out = out.Merge(lbs)
 		}
-		t.client.Chan() <- api.Entry{
-			Entry: logproto.Entry{
-				Line:      string(message.Value),
-				Timestamp: timestamp(t.useIncomingTimestamp, message.Timestamp),
-			},
-			Labels: out,
+
+		logLines := []string{string(message.Value)}
+
+		if t.messageParser != nil {
+			ll, err := t.messageParser(message)
+			if err != nil {
+				level.Error(t.logger).Log("msg", "message parsing error", "err", err)
```

Comment:
if there's an error parsing, I'd skip the current kafka message logging the error, that's the usual approach in promtail

Reply:
Sounds good 👍 Changed.